### PR TITLE
Updates phpstan version and fixes newly discovered bugs

### PR DESCRIPTION
**Pull request by bytehead**

### CODE REVIEW

File: core-bundle/src/Command/DoctrineMigrationsDiffCommand.php [hunk 1]
```diff
@@ -30,7 +30,7 @@ class DoctrineMigrationsDiffCommand extends DiffCommand
     /**
      * {@inheritdoc}
      */
-    public function execute(InputInterface $input, OutputInterface $output): void
+    public function execute(InputInterface $input, OutputInterface $output): ?int
```
Comment (bytehead) [inline core-bundle/src/Command/DoctrineMigrationsDiffCommand.php]: See here: https://github.com/symfony/console/blob/master/Command/Command.php#L153

File: core-bundle/src/Csrf/MemoryTokenStorage.php [hunk 2]
```diff
@@ -67,7 +67,7 @@ public function hasToken($tokenId): bool
     /**
      * {@inheritdoc}
      */
-    public function removeToken($tokenId): void
+    public function removeToken($tokenId): ?string
```
Comment (bytehead) [inline core-bundle/src/Csrf/MemoryTokenStorage.php]: See here: https://github.com/symfony/security/blob/master/Csrf/TokenStorage/TokenStorageInterface.php#L48

File: core-bundle/src/Csrf/MemoryTokenStorage.php [hunk 3]
```diff
@@ -67,12 +67,19 @@ public function hasToken($tokenId): bool
...
     {
         $this->assertInitialized();
 
+        $token = null;
         $this->usedTokens[$tokenId] = true;
-        $this->tokens[$tokenId] = null;
+
+        if (array_key_exists($tokenId, $this->tokens)) {
```
Comment (leofeyer) [inline core-bundle/src/Csrf/MemoryTokenStorage.php]: Why did you add this?

File: composer.json [hunk 4]
```diff
@@ -132,15 +132,15 @@
...
+        "contao/phpstan": "^0.5",
         "contao/test-case": "^2.0",
         "doctrine/doctrine-migrations-bundle": "^1.1",
         "doctrine/event-manager": "^1.0",
         "doctrine/orm": "^2.5",
         "friendsofphp/php-cs-fixer": "^2.12",
         "monolog/monolog": "^1.22",
         "php-coveralls/php-coveralls": "^2.1",
-        "phpstan/phpstan-phpunit": "^0.10",
+        "phpstan/phpstan-phpunit": "^0.11",
```
Comment (leofeyer) [inline composer.json:143]: Why is this necessary?

**Edit:** nvm, I found out myself. 😁

Comment (bytehead) [reply composer.json:143]: https://getcomposer.org/doc/articles/versions.md#caret-version-range-

> The ^ operator behaves very similarly but it sticks closer to semantic versioning, and will always allow non-breaking updates. For example ^1.2.3 is equivalent to >=1.2.3 <2.0.0 as none of the releases until 2.0 should break backwards compatibility. For pre-1.0 versions it also acts with safety in mind and treats ^0.3 as >=0.3.0 <0.4.0.

File: core-bundle/src/Csrf/MemoryTokenStorage.php (hunk 3 above)
Comment (leofeyer) [reply core-bundle/src/Csrf/MemoryTokenStorage.php]: @aschempp @ausi According to the [interface](https://github.com/symfony/security/blob/master/Csrf/TokenStorage/TokenStorageInterface.php#L45), we are supposed to return `string|null`. However, we are not returning anything at the moment.

1. How can this have worked?
2. Are the changes correct in your opinion?

**Comment by leofeyer** [comment]
Since intersection type support is still buggy in PhpStorm (see https://youtrack.jetbrains.com/issue/WI-45275), I would prefer not to merge this into the `4.7` branch. WDYT?

File: core-bundle/src/Csrf/MemoryTokenStorage.php (hunk 3 above)
Comment (ausi) [reply core-bundle/src/Csrf/MemoryTokenStorage.php]: > 1. How can this have worked?

Returning nothing is the same as `null` so there was no type error. We don’t use the return value of this method in our code so it probably didn’t have an effect for our use case.

> 2. Are the changes correct in your opinion?

Yes, not returing the token was a mistake :(  
We should probably test the return value in the unit test: https://github.com/contao/contao/blob/b71079a3b55d55950d87cd972e1d0f865c121d89/core-bundle/tests/Csrf/MemoryTokenStorageTest.php#L36-L41

Comment (aschempp) [reply core-bundle/src/Csrf/MemoryTokenStorage.php]: Completely agree with @ausi

Comment (leofeyer) [reply core-bundle/src/Csrf/MemoryTokenStorage.php]: I have backported the fixes in 66803d67f0b3069470107438f86d2c8171ec132d.

**Comment by leofeyer** [comment]
Bug report for the EA Extended plugin: https://github.com/kalessil/phpinspectionsea/issues/1321

**Comment by leofeyer** [comment]
Implemented in 388677ec0a9083360488307eb6cefc1ff96a99d8 for Contao 4.8. Thank you @bytehead.
